polaris-core/src/main/java/org/apache/polaris/core/policy/content/maintenance/OrphanFileRemovalPolicyContent.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.polaris.core.policy.content.maintenance;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Strings;
+import java.util.List;
+import java.util.Set;
+import org.apache.polaris.core.policy.content.PolicyContentUtil;
+import org.apache.polaris.core.policy.validator.InvalidPolicyException;
+
+public class OrphanFileRemovalPolicyContent extends BaseMaintenancePolicyContent {
+  private static final String DEFAULT_POLICY_SCHEMA_VERSION = "2025-02-03";
+  private static final Set<String> POLICY_SCHEMA_VERSIONS = Set.of(DEFAULT_POLICY_SCHEMA_VERSION);
+
+  @JsonProperty(value = "max_orphan_file_age_in_days")
+  private int maxOrphanFileAgeInDays;
+
+  private List<String> locations;
+
+  @JsonCreator
+  public OrphanFileRemovalPolicyContent(
+      @JsonProperty(value = "enable", required = true) boolean enable) {
+    super(enable);
+  }
+
+  public int getMaxOrphanFileAgeInDays() {
+    return maxOrphanFileAgeInDays;
+  }
+
+  public void setMaxOrphanFileAgeInDays(int maxOrphanFileAgeInDays) {
+    this.maxOrphanFileAgeInDays = maxOrphanFileAgeInDays;
+  }
+
+  public List<String> getLocations() {
+    return locations;
+  }
+
+  public void setLocations(List<String> locations) {
+    this.locations = locations;
+  }
+
+  public static OrphanFileRemovalPolicyContent fromString(String content) {
+    if (Strings.isNullOrEmpty(content)) {
```

Comment:
Instead of repeating this everywhere, perhaps we can push this check up into the caller -- `PolicyValidators` is a good narrow waist

Reply:
I think it'd nice to keep this parsing method in each individual class, so that they can be invoked separately, not only for validation use cases, but also for a general parsing or constructor  use cases. WDYT?

Reply:
Making sure that they can be invoked separately makes sense. Maybe a supertype could contain some of this logic then?

Reply:
Was trying to do that per your comment here, https://github.com/apache/polaris/pull/1238#discussion_r2012627006. `fromString` are static methods, which cannot be overridden. The only thing we could do for static functions is to having another static function to group common logic. Let me refactor a bit.